### DISCUSSION TITLE
Unable to connect GNB AND UE, Error (Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=1134)

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#169] [General Help]
Hello everyone,

I am in the process of implementing a Standalone (SA) project for gNodeB (GNB). I am utilizing a USRP B210 for the User Equipment (UE) and a USRP B200 for the GNB. I have included a Google Drive link that contains all the log files and pcap files. Your review would be greatly appreciated.

Previously, I faced an issue related to "Tx while waiting for EOB". However, by adjusting the CFO to -5200 and configuring the TX and RX gains, I succeeded in establishing the RRC connection.

Within the log, I observed the message "PBCH-MIB: CRC failed". Could someone kindly provide insights on how to address this issue?

In addition, I have a couple of questions:

Q1) Currently, I am using the internal clock. Many suggestions recommend acquiring an external clock to resolve this error. Do I need to purchase an external clock, or can I continue using the internal clock?

Q2) What types of USRP antennas are recommended for establishing a connection between the GNB and UE without requiring an external clock?

Thank you very much.

**

**### UE:-**

**

raju@raju-ThinkPad-P16s-Gen-1:~/srsRAN_4G/build/srsue/src$ sudo ./srsue /home/raju/Downloads/ue_rf.conf_org
Active RF plugins: libsrsran_rf_uhd.so libsrsran_rf_zmq.so
Inactive RF plugins: 
Reading configuration file /home/raju/Downloads/ue_rf.conf_org...

Built in Release mode using commit fa56836b1 on branch master.

Opening 1 channels in RF device=uhd with args=clock=internal
Supported RF device list: UHD zmq file
[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[INFO] [LOGGING] Fastpath logging disabled at runtime.
[INFO] [B200] Loading firmware image: /usr/local/share/uhd/images/usrp_b200_fw.hex...
Opening USRP channels=1, args: type=b200,master_clock_rate=23.04e6
[INFO] [UHD RF] RF UHD Generic instance constructed
[INFO] [B200] Detected Device: B210
[INFO] [B200] Loading FPGA image: /usr/local/share/uhd/images/usrp_b210_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Detecting internal GPSDO.... 
[INFO] [GPS] No GPSDO found
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Asking for clock rate 23.040000 MHz... 
[INFO] [B200] Actually got clock rate 23.040000 MHz.
Setting manual TX/RX offset to 30 samples
Waiting PHY to initialize ... done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=3214
Random Access Complete.     c-rnti=0x4604, ta=45
RRC Connected
Scheduling request failed: releasing RRC connection...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=5934
Random Access Complete.     c-rnti=0x4604, ta=45
RF status: O=0, U=1, L=0
RF status: O=0, U=1, L=0
RF status: O=0, U=2, L=0
RF status: O=0, U=1, L=0
RF status: O=0, U=1, L=0
RF status: O=0, U=1, L=0
RF status: O=0, U=2, L=0
RF status: O=0, U=1, L=0
RF status: O=0, U=1, L=0
Scheduling request failed: releasing RRC connection...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=2734
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=2894
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=3054
RF status: O=0, U=1, L=0
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=3214
RF status: O=0, U=1, L=0
**### 

> GNB:-

**
raju@raju-ThinkPad-P16s-Gen-1:~/srsRAN_Project/build/apps/gnb$ sudo ./gnb -c /home/raju/srsRAN_Project/configs/gnb_rf_b210_fdd_srsUE.yml
Lower PHY in quad executor mode.

--== srsRAN gNB (commit 1afd7240f) ==--

Connecting to AMF on 127.0.1.100:38412
Failed to bind UDP socket to 127.0.0.1:2152. Address already in use
Available radio types: uhd.
Making USRP object with args 'type=b200'
[INFO] [UHD] linux; GNU C++ version 11.4.0; Boost_107400; UHD_4.4.0.0-265-ga2a04e31
[INFO] [LOGGING] Fastpath logging disabled at runtime.
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test...
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz...
[INFO] [B200] Actually got clock rate 16.000000 MHz.
[INFO] [MULTI_USRP] Setting master clock rate selection to 'manual'.
[INFO] [B200] Asking for clock rate 11.520000 MHz...
[INFO] [B200] Actually got clock rate 11.520000 MHz.
Cell pci=1, bw=10 MHz, dl_arfcn=368500 (n3), dl_freq=1842.5 MHz, dl_ssb_arfcn=368410, ul_freq=1747.5 MHz

==== gNodeB started ===
Type <t> to view trace
Late: 0; Underflow: 2; Overflow: 0;
Late: 6; Underflow: 3; Overflow: 0;
Late: 0; Underflow: 3; Overflow: 0;
Late: 0; Underflow: 3; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 12; Underflow: 4; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;

https://drive.google.com/drive/folders/10J-RzKFBoh66PtkGU_AiX9zEW429HydW?usp=drive_link

**Comment by pgawlowicz** [comment]
Hi, Please follow the updated tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

In addition, please run the following script on both PCs:
https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance
